### PR TITLE
Add, Mul, and Sum

### PR DESCRIPTION
**Pull request by varunjain99**
Translation for Add and Mul operators. Refactoring modifies Sum translation
@bddppq @yinghai @Maratyszcza

### CODE REVIEW

File: onnx_xla/operator_registry.cc [hunk 1]
```diff
@@ -34,4 +34,21 @@ TranslationMap& OperatorRegistry::map() {
...
+    const XlaBuilder& builder,
+    const XlaOp& firstOp,
+    const XlaOp& secondOp) {
+  auto firstNDim = ShapeUtil::Rank(builder.GetShape(firstOp).ValueOrDie());
+  auto secondNDim = ShapeUtil::Rank(builder.GetShape(secondOp).ValueOrDie());
+  std::vector<int64> broadcastDims;
+  if (firstNDim != secondNDim || firstNDim != 0 || secondNDim != 0) {
+    auto minDim = std::min(firstNDim, secondNDim);
+    auto maxDim = std::max(firstNDim, secondNDim);
+    for (auto j = 0; j < minDim; ++j) {
```
Comment (bddppq) [inline onnx_xla/operator_registry.cc]: nit: std::iota

File: onnx_xla/translations/translate_add.cc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#include "onnx_xla/operator_registry.h"
```
Comment (bddppq) [inline onnx_xla/translations/translate_add.cc]: all the elementwise binary operators are similar, maybe use a macro?

File: onnx_xla/operator_registry.cc (hunk 1 above)
Comment (varunjain99) [reply onnx_xla/operator_registry.cc]: Have to resize first and then use iota so it does two passes?

File: onnx_xla/translations/translate_add.cc (hunk 2 above)
Comment (varunjain99) [reply onnx_xla/translations/translate_add.cc]: Yeah makes sense with Sub and Div - changed

File: onnx_xla/operator_registry.cc (hunk 1 above)
Comment (yinghai) [reply onnx_xla/operator_registry.cc]: Yes, `resize` pads the space and `iota` fills in data until it hits the end of the iterator. You need both. But here you can just push. No need to go 2 pass.

**Pull request merged**